### ISSUE TITLE
Error during Layout/EmptyLineAfterGuardClause

### ISSUE THREAD

**Issue by forthrin** [rubocop/rubocop#14075]
## Steps to reproduce the problem

```bash
echo "if 1 + 1 then return end.then { 2 }" > foo.rb
rubocop -c /dev/null foo.rb
```

## Actual behavior

```
An error occurred while Layout/EmptyLineAfterGuardClause cop was inspecting foo.rb:1:0.
Errors are usually caused by RuboCop bugs.
Please, update to the latest RuboCop version if not already in use, and report a bug if the issue still occurs on this version.
https://github.com/rubocop/rubocop/issues
```

## RuboCop version

```
Mention the following information in the issue report:
1.75.2 (using Parser 3.3.7.4, rubocop-ast 1.44.0, analyzing as Ruby 2.7, running on ruby 3.4.2) [arm64-darwin24]
```
